cryptolib/signer.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import "errors"
+
+// Signer contains methods to create a signed Attestation.
+type Signer interface {
+	CreateAttestation(privateKey, payload []byte, publicKeyID string, alg SigningAlgorithm, keyMode KeyMode) (*Attestation, error)
+}
+
+type signer struct{}
+
+// CreateAttestation creates an Attestation whose signature is generated by
+// signing the given payload with the private key. For PGP and PKIX, `payload`
+// should be the raw payload data. For JWT, `payload` should be a serialized
+// but unsigned token.
+func (s *signer) CreateAttestation(privateKey, payload []byte, publicKeyID string, alg SigningAlgorithm, keyMode KeyMode) (*Attestation, error) {
```

Comment:
Just same comment as above, depending on how `signer` is instantiated. 
An alternative to this is having `CreateAttestation` as a method in the interface, and separating implemented by each Signer type.
Not suggesting this is the way to do it, but just throwing out an alternative here.